### PR TITLE
[#2156] improvement: enable JavaUtilDate error-prone check

### LINKED ISSUE #2156: [Improvement] Improve use of JavaUtilDate in KerberosTokenProvider.java
### What would you like to be improved?

KerberosTokenProvider.java:122: warning: [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.

```java
return tickets.iterator().next().getEndTime().getTime() < System.currentTimeMillis();
```

See https://errorprone.info/bugpattern/JavaUtilDate

### How should we improve?

Consider improving this

### PR DESCRIPTION
**Pull request by coolderli**
### What changes were proposed in this pull request?

- close #2156 

### Why are the changes needed?

Fix: #2156 

### Does this PR introduce _any_ user-facing change?

- no

### How was this patch tested?

- `./gradlew build`

### CODE REVIEW

**Comment by coolderli** [comment]
`JavaUtilDate` is to avoid `java.util.Date`. Because `java.util.Date` has a bad API that leads to bugs; prefer `java.time.Instant `or `LocalDate`. More details can be found here: https://errorprone.info/bugpattern/JavaUtilDate.

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/KerberosTokenProvider.java [hunk 1]
```diff
@@ -113,6 +113,7 @@ public byte[] call() throws Exception {
         });
   }
 
+  @SuppressWarnings("JavaUtilDate")
```
Comment (coolderli) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/KerberosTokenProvider.java:116]: `java.util.Date` comes form `javax.security.auth.kerberos.KerberosTicket.getEndTime()`. We can't avoid it. So just ignore it.

```
/Users/dian/Project/gravitino/clients/client-java/src/main/java/com/datastrato/gravitino/client/KerberosTokenProvider.java:124: 警告: [JavaUtilDate] Date has a bad A that leads to bugs; prefer java.time.Instant or LocalDate.
    return tickets.iterator().next().getEndTime().getTime() < System.currentTimeMillis();
                                                         ^
    (see https://errorprone.info/bugpattern/JavaUtilDate)

```

File: clients/client-java/src/test/java/com/datastrato/gravitino/client/TestOAuth2TokenProvider.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
 import org.mockserver.matchers.Times;
 import org.mockserver.model.HttpResponse;
 
+@SuppressWarnings("JavaUtilDate")
```
Comment (coolderli) [inline clients/client-java/src/test/java/com/datastrato/gravitino/client/TestOAuth2TokenProvider.java:34]: `Date` comes from `io.jsonwebtoken.JwtBuilder#setExpiration`. I can't figure out a way to fix it. So I just ignore it.

```
/Users/dian/Project/gravitino/clients/client-java/src/test/java/com/datastrato/gravitino/client/TestOAuth2TokenProvider.java:126: 警告: [JavaUtilDate] Date has a badPI that leads to bugs; prefer java.time.Instant or LocalDate.
            .setExpiration(new Date(System.currentTimeMillis() + 10000))
                           ^
    (see https://errorprone.info/bugpattern/JavaUtilDate)

```

File: server-common/src/test/java/com/datastrato/gravitino/server/auth/TestOAuth2TokenAuthenticator.java [hunk 3]
```diff
@@ -22,6 +22,7 @@
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
+@SuppressWarnings("JavaUtilDate")
```
Comment (coolderli) [inline server-common/src/test/java/com/datastrato/gravitino/server/auth/TestOAuth2TokenAuthenticator.java:25]: Same as above.

**Comment by coolderli** [comment]
@justinmclean Please help review this when you have time. Thanks.

**Comment by jerryshao** [comment]
@qqqttt123 please help to review this.

**Comment by qqqttt123** [comment]
> @qqqttt123 please help to review this.

OK.

**Comment by qqqttt123** [comment]
I feel a little confused about this. We add a error-prone check. But we ignore them. It seems a little weird. Do the some methods have alternative method. We can the alternative method.

**Comment by coolderli** [comment]
> I feel a little confused about this. We add a error-prone check. But we ignore them. It seems a little weird. Do the some methods have alternative method. We can the alternative method.

@qqqttt123 I'm not sure if there is another SDK. I think we should not change it for error-prone. 
After we add this, we can avoid using `java.util.Date` in the future. For example, if we use `new Date()` in our code, it will report an error. If we have to use it we can ignore it.

**Review by qqqttt123 — APPROVED** [review]
OK, LGTM.

**Review by jerryshao — APPROVED** [review]
LGTM. Thanks @coolderli for your contribution.

**Pull request merged**